### PR TITLE
add recovery guide for two dc deployment

### PR DESCRIPTION
**Pull request by disksing**
Signed-off-by: disksing <i@disksing.com>

### What is changed, added or deleted? (Required)

- add description of the `reset-to-version` command of `tikv-ctl`
- add detailed steps to recover data for dual dc deployment

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v6.2 (TiDB 6.2 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Review by lhy1024 — APPROVED** [review]

**Review by longfangsong — APPROVED** [review]

**Review by tshqin — APPROVED** [review]
LGTM

**Comment by disksing** [comment]
The feature has serious issue. Closing for v6.0.

File: two-data-centers-in-one-city-deployment.md [hunk 1]
```diff
@@ -275,14 +275,30 @@ curl http://pd_ip:pd_port/pd/api/v1/replication_mode/status
...
-- 如果丢失了大多数 Voter 副本，需要人工介入，通过专业工具恢复（恢复方式请联系 TiDB 团队）。
+3. 如果故障前处于异步切换同步状态（状态码为 sync-recover），使用从中心数据恢复服务后，会丢失一部分主集群在异步模式下写入的数据，并且可能会发生数据 ACID 不一致的情况，需要进行额外的 ACID 一致性恢复操作。这种情况的典型场景是，先发生了主从断连，切换至异步模式写入一些数据后，主从中心连接恢复，在主从中心同步数据的过程中，再次发生故障，并导致主中心整体不可用。
+
+恢复流程（如果需要恢复数据技术支持，请联系 TiDB 团队）：
+
+1. 停止从中心所有的 PD / TiKV / TiDB 服务。
+
+2. 添加 `--force-new-cluster` 参数，以单副本模式启动从中心的 PD 节点。
+
+3. 使用 [Online Unsafe Recovery](/online-unsafe-recovery) 处理从中心的 TiKV 数据，参数为主中心所有 Store ID 的列表。
```
Comment (Oreoxmt) [inline two-data-centers-in-one-city-deployment.md:296]:
```suggestion
3. 使用 [Online Unsafe Recovery](/online-unsafe-recovery.md) 处理从中心的 TiKV 数据，参数为主中心所有 Store ID 的列表。
```

Comment (Oreoxmt) [inline two-data-centers-in-one-city-deployment.md:296]:
```suggestion
3. 使用 [Online Unsafe Recovery](/online-unsafe-recovery.md) 处理从中心的 TiKV 数据，参数为主中心所有 Store ID 的列表。
```

**Comment by Oreoxmt** [comment]
@longfangsong @lhy1024 @tshqin PTAL

**Review by longfangsong — APPROVED** [review]

**Review by lhy1024 — APPROVED** [review]

File: two-data-centers-in-one-city-deployment.md [hunk 2]
```diff
@@ -275,14 +275,30 @@ curl http://pd_ip:pd_port/pd/api/v1/replication_mode/status
...
 
-当不处于同步复制状态的集群发生了灾难，不能保证满足 `RPO = 0` 进行数据恢复：
+2. 如果故障前处于异步复制状态（状态码为 async），使用从中心数据恢复服务后，会丢失主集群在异步模式下写入的数据。这种情况的典型场景是，主集群完全故障前，先发生了主从断连，且主集群切换至异步复制状态并提供了一段时间服务。
 
-- 如果丢失了大多数 Voter 副本，需要人工介入，通过专业工具恢复（恢复方式请联系 TiDB 团队）。
+3. 如果故障前处于异步切换同步状态（状态码为 sync-recover），使用从中心数据恢复服务后，会丢失一部分主集群在异步模式下写入的数据，并且可能会发生数据 ACID 不一致的情况，需要进行额外的 ACID 一致性恢复操作。这种情况的典型场景是，先发生了主从断连，切换至异步模式写入一些数据后，主从中心连接恢复，在主从中心同步数据的过程中，再次发生故障，并导致主中心整体不可用。
+
+恢复流程（如果需要恢复数据技术支持，请联系 TiDB 团队）：
+
+1. 停止从中心所有的 PD / TiKV / TiDB 服务。
```
Comment (Oreoxmt) [inline two-data-centers-in-one-city-deployment.md:292]:
```suggestion
1. 停止从中心所有的 PD、TiKV、TiDB 服务。
```

File: two-data-centers-in-one-city-deployment.md [hunk 3]
```diff
@@ -275,14 +275,30 @@ curl http://pd_ip:pd_port/pd/api/v1/replication_mode/status
...
+
+2. 添加 `--force-new-cluster` 参数，以单副本模式启动从中心的 PD 节点。
+
+3. 使用 [Online Unsafe Recovery](/online-unsafe-recovery) 处理从中心的 TiKV 数据，参数为主中心所有 Store ID 的列表。
+
+4. 使用 [PD Control 工具](/pd-control.md) 写入新的 placement rule 配置，其中 Region 的 Voter 副本数配置为原集群在从中心的副本数。
+
+5. 启动 PD 和 TiKV 服务。
+
+6. 如果需要进行 ACID 一致性恢复（原 PD 的 DR_STATE 文件内状态为 sync-recover），使用 [TiKV Control 工具的 reset-to-version](/tikv-control.md#恢复-acid-不一致的数据) 处理 TiKV 数据，所使用的 version 参数可以使用 `pd-ctl min-resolved-ts`查得。
```
Comment (Oreoxmt) [inline two-data-centers-in-one-city-deployment.md:302]:
```suggestion
6. 如果需要进行 ACID 一致性恢复（原 PD 的 DR_STATE 文件内状态为 sync-recover），使用 [TiKV Control 工具的 `reset-to-version`](/tikv-control.md#恢复-acid-不一致的数据) 处理 TiKV 数据，所使用的 version 参数可以使用 `pd-ctl min-resolved-ts` 查得。
```

File: two-data-centers-in-one-city-deployment.md [hunk 4]
```diff
@@ -275,14 +275,30 @@ curl http://pd_ip:pd_port/pd/api/v1/replication_mode/status
...
+本节介绍同城两中心部署提供的容灾恢复方案。本节所讨论的灾难是指主中心发生整体故障，或者主/从中心多台 TiKV 节点发生故障，导致数据或一部分数据丢失多数副本而无法提供服务的情形。
 
-当处于同步复制状态的集群发生了灾难，可进行 `RPO = 0` 的数据恢复：
+#### 主中心整体故障
 
-- 如果主数据中心发生故障，丢失了大多数 Voter 副本，但是从数据中心有完整的数据，可在从数据中心恢复数据。此时需要人工介入，通过专业工具恢复（恢复方式请联系 TiDB 团队）。
+这种情况下，所有 Region 都失去了大多数副本，因而集群无法提供服务，我们需要使用从中心的数据来恢复服务，恢复的能力根据故障前的复制状态决定：
 
-- 如果从数据中心发生故障，丢失了少数 Voter 副本，能自动切换成 async 异步复制模式。
+1. 如果故障前处于同步复制状态（状态码为 sync 或 async_wait），可使用从中心数据进行 `RPO = 0` 的数据恢复。
```
Comment (Oreoxmt) [inline two-data-centers-in-one-city-deployment.md:284]:
```suggestion
- 如果故障前处于同步复制状态（状态码为 `sync` 或 `async_wait`），可使用从中心数据进行 `RPO = 0` 的数据恢复。
```

File: tikv-control.md [hunk 5]
```diff
@@ -551,6 +551,28 @@ success!
 > - `-p` 选项指定 PD 的 endpoint，不使用 `http` 前缀，用于查询指定的 `region_id` 是否有效。
 > - 对于指定 Region 的 peers 所在的每个 store，均须执行该命令。
 
+### 恢复 ACID 不一致的数据
+
+`reset-to-version` 命令用于数据丢失大部分副本、数据同步不完整等问题导致出现数据包含不完整事务的情况。此时需要提供一个更旧的能保证 ACID 一致性的版本号。`tikv-ctl` 会清除掉所有版本号更大的数据。
```
Comment (TomShawn) [inline tikv-control.md:556]: 能否说明是什么版本呢？以及如何获取呢？
```suggestion
在集群上，如果出现了数据丢失大部分副本或数据同步不完整等问题，从而导致出现数据包含不完整事务的情况，你可以使用 `reset-to-version` 命令用于恢复 ACID 不一致的数据。此时你需要提供一个能保证 ACID 一致性的旧版本号。`tikv-ctl` 会清除掉在此版本之后的所有数据。
```

File: tikv-control.md [hunk 6]
```diff
@@ -551,6 +551,28 @@ success!
...
+```
+
+```
+success!
+```
+
+> **注意：**
+>
+> - 该命令只支持在线模式，且应该停止 TiDB 等会向 TiKV 写入数据的进程。在运行成功后，会打印 `success!`。
+> - 集群中所有 TiKV 需要执行相同的命令。
```
Comment (TomShawn) [inline tikv-control.md:573]:
```suggestion
> - 你需要对集群中所有的 TiKV 节点执行相同的命令。
```

File: tikv-control.md [hunk 7]
```diff
@@ -551,6 +551,28 @@ success!
...
+tikv-ctl --host 127.0.0.1:20160 reset-to-version -v 430315739761082369
+```
+
+```
+success!
+```
+
+> **注意：**
+>
+> - 该命令只支持在线模式，且应该停止 TiDB 等会向 TiKV 写入数据的进程。在运行成功后，会打印 `success!`。
```
Comment (TomShawn) [inline tikv-control.md:572]:
```suggestion
> - 该命令只支持在线模式。在执行命令前，你需要停止 TiDB 等会向 TiKV 写入数据的进程。在命令执行成功后，会返回 `success!`。
```

File: two-data-centers-in-one-city-deployment.md [hunk 8]
```diff
@@ -275,14 +275,32 @@ curl http://pd_ip:pd_port/pd/api/v1/replication_mode/status
...
 ### 灾难恢复
 
-本节介绍同城两中心部署提供的容灾恢复方案。
+本节介绍同城两中心部署提供的容灾恢复方案。本节所讨论的灾难是指主中心发生整体故障，或者主/从中心多台 TiKV 节点发生故障，导致数据或一部分数据丢失多数副本而无法提供服务的情形。
 
-当处于同步复制状态的集群发生了灾难，可进行 `RPO = 0` 的数据恢复：
+#### 主中心整体故障
 
-- 如果主数据中心发生故障，丢失了大多数 Voter 副本，但是从数据中心有完整的数据，可在从数据中心恢复数据。此时需要人工介入，通过专业工具恢复（恢复方式请联系 TiDB 团队）。
+这种情况下，所有 Region 都失去了大多数副本，因而集群无法提供服务，我们需要使用从中心的数据来恢复服务，恢复的能力根据故障前的复制状态决定：
```
Comment (TomShawn) [inline two-data-centers-in-one-city-deployment.md:282]:
```suggestion
这种情况下，所有 Region 都失去了大多数副本，因而集群无法提供服务。此时需要使用从中心的数据来恢复服务，恢复的能力由故障前的复制状态决定：
```

File: two-data-centers-in-one-city-deployment.md [hunk 9]
```diff
@@ -275,14 +275,32 @@ curl http://pd_ip:pd_port/pd/api/v1/replication_mode/status
...
+- 如果故障前处于异步复制状态（状态码为 `async`），使用从中心数据恢复服务后，会丢失主集群在异步模式下写入的数据。这种情况的典型场景是，主集群完全故障前，先发生了主从断连，且主集群切换至异步复制状态并提供了一段时间服务。
 
-- 如果丢失了大多数 Voter 副本，需要人工介入，通过专业工具恢复（恢复方式请联系 TiDB 团队）。
+- 如果故障前处于异步切换同步状态（状态码为 `sync-recover`），使用从中心数据恢复服务后，会丢失一部分主集群在异步模式下写入的数据，并且可能会发生数据 ACID 不一致的情况，需要进行额外的 ACID 一致性恢复操作。这种情况的典型场景是，先发生了主从断连，切换至异步模式写入一些数据后，主从中心连接恢复，在主从中心同步数据的过程中，再次发生故障，并导致主中心整体不可用。
+
+恢复流程：
+
+1. 停止从中心所有的 PD、TiKV、TiDB 服务。
+
+2. 添加 `--force-new-cluster` 参数，以单副本模式启动从中心的 PD 节点。
```
Comment (TomShawn) [inline two-data-centers-in-one-city-deployment.md:294]: 是在 tikv-ctl 中添加吗？

File: two-data-centers-in-one-city-deployment.md [hunk 10]
```diff
@@ -275,14 +275,32 @@ curl http://pd_ip:pd_port/pd/api/v1/replication_mode/status
...
+
+3. 使用 [Online Unsafe Recovery](/online-unsafe-recovery.md) 处理从中心的 TiKV 数据，参数为主中心所有 Store ID 的列表。
+
+4. 使用 [PD Control 工具](/pd-control.md) 写入新的 placement rule 配置，其中 Region 的 Voter 副本数配置为原集群在从中心的副本数。
+
+5. 启动 PD 和 TiKV 服务。
```
Comment (TomShawn) [inline two-data-centers-in-one-city-deployment.md:300]:
```suggestion
5. 启动主集群的 PD 和 TiKV 服务。
```

File: two-data-centers-in-one-city-deployment.md [hunk 11]
```diff
@@ -275,14 +275,32 @@ curl http://pd_ip:pd_port/pd/api/v1/replication_mode/status
...
+
+6. 如果需要进行 ACID 一致性恢复（原 PD 的 `DR_STATE` 文件内状态为 `sync-recover`），使用 [TiKV Control 工具的 `reset-to-version`](/tikv-control.md#恢复-acid-不一致的数据) 处理 TiKV 数据。所使用的 `version` 参数可以使用 `pd-ctl min-resolved-ts` 查得。
+
+7. 开启 TiDB 服务，检查数据的完整性和一致性。
```
Comment (TomShawn) [inline two-data-centers-in-one-city-deployment.md:304]:
```suggestion
7. 开启主集群的 TiDB 服务，检查数据的完整性和一致性。
```

**Review by TomShawn — COMMENTED** [review]
整体看下来，感觉蛮怕怕。是否可以说明下，建议联系 TiDB 团队呢？

File: two-data-centers-in-one-city-deployment.md (hunk 9 above)
Comment (disksing) [reply two-data-centers-in-one-city-deployment.md:294]: pd-server的启动参数

File: two-data-centers-in-one-city-deployment.md [hunk 12]
```diff
@@ -275,14 +275,32 @@ curl http://pd_ip:pd_port/pd/api/v1/replication_mode/status
...
+- 如果故障前处于异步复制状态（状态码为 `async`），使用从中心数据恢复服务后，会丢失主集群在异步模式下写入的数据。这种情况的典型场景是，主集群完全故障前，先发生了主从断连，且主集群切换至异步复制状态并提供了一段时间服务。
 
-- 如果丢失了大多数 Voter 副本，需要人工介入，通过专业工具恢复（恢复方式请联系 TiDB 团队）。
+- 如果故障前处于异步切换同步状态（状态码为 `sync-recover`），使用从中心数据恢复服务后，会丢失一部分主集群在异步模式下写入的数据，并且可能会发生数据 ACID 不一致的情况，需要进行额外的 ACID 一致性恢复操作。这种情况的典型场景是，先发生了主从断连，切换至异步模式写入一些数据后，主从中心连接恢复，在主从中心同步数据的过程中，再次发生故障，并导致主中心整体不可用。
+
+恢复流程：
+
+1. 停止从中心所有的 PD、TiKV、TiDB 服务。
+
+2. 添加 `--force-new-cluster` 参数，以单副本模式启动从中心的 PD 节点。
```
Comment (Oreoxmt) [inline two-data-centers-in-one-city-deployment.md:294]:
```suggestion
2. 启动 PD 时添加 [`--force-new-cluster`](/command-line-flags-for-pd-configuration.md#--force-new-cluster) 参数，以单副本模式启动从中心的 PD 节点。
```

File: two-data-centers-in-one-city-deployment.md [hunk 13]
```diff
@@ -275,14 +275,32 @@ curl http://pd_ip:pd_port/pd/api/v1/replication_mode/status
...
 ### 灾难恢复
 
-本节介绍同城两中心部署提供的容灾恢复方案。
+本节介绍同城两中心部署提供的容灾恢复方案。本节所讨论的灾难是指主中心发生整体故障，或者主/从中心多台 TiKV 节点发生故障，导致数据或一部分数据丢失多数副本而无法提供服务的情形。
 
-当处于同步复制状态的集群发生了灾难，可进行 `RPO = 0` 的数据恢复：
+#### 主中心整体故障
 
-- 如果主数据中心发生故障，丢失了大多数 Voter 副本，但是从数据中心有完整的数据，可在从数据中心恢复数据。此时需要人工介入，通过专业工具恢复（恢复方式请联系 TiDB 团队）。
+这种情况下，所有 Region 都失去了大多数副本，因而集群无法提供服务。此时需要使用从中心的数据来恢复服务，恢复的能力由故障前的复制状态决定：
```
Comment (TomShawn) [inline two-data-centers-in-one-city-deployment.md:282]:
```suggestion
这种情况下，主中心的所有 Region 都失去了大多数副本，因而集群无法提供服务。此时需要使用从中心的数据来恢复服务，恢复的能力由故障前的复制状态决定：
```

File: tikv-control.md [hunk 14]
```diff
@@ -551,6 +551,28 @@ success!
 > - `-p` 选项指定 PD 的 endpoint，不使用 `http` 前缀，用于查询指定的 `region_id` 是否有效。
 > - 对于指定 Region 的 peers 所在的每个 store，均须执行该命令。
 
+### 恢复 ACID 不一致的数据
+
+在集群上，如果出现了数据丢失大部分副本或数据同步不完整等问题，从而导致出现数据包含不完整事务的情况，你可以使用 `reset-to-version` 命令用于恢复 ACID 不一致的数据。此时你需要提供一个能保证 ACID 一致性的旧版本号。`tikv-ctl` 会清除掉在此版本之后的所有数据。
+
+- `-v` 选项用于指定恢复的版本号。
```
Comment (TomShawn) [inline tikv-control.md:558]: 这里可以提一下「所使用的 `version` 参数可以使用 `pd-ctl min-resolved-ts` 查得」吗？

Comment (disksing) [inline tikv-control.md:558]:
```suggestion
- `-v` 选项用于指定恢复的版本号。所使用的 `-v` 参数可以使用 `pd-ctl min-resolved-ts` 查得。
```

**Review by Oreoxmt — APPROVED** [review]
LGTM

**Pull request merged**